enumerables.rb

```diff
@@ -0,0 +1,175 @@
+# module custom Enumerables
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  # My Each Method
+
+  def my_each
+    return to_enum unless block_given?
+
+    length.times do |i|
+      yield(self[i])
+    end
+  end
+
+  # My  Each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    length.times do |i|
+      yield(i, self[i])
+    end
+  end
+
+  # My Select Method
+  def my_select
+    return to_enum unless block_given?
+
+    invited_list = []
+    my_each do |friend|
+      invited_list.push(friend) if yield(friend)
+    end
+
+    invited_list
+  end
+
+  # rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+  # My  All
+  def my_all?(args = nil)
+    if block_given?
+      my_each do |element|
+        return false if yield(element) == false
+      end
+    elsif args.nil?
+      my_each do |i|
+        return false if i.nil?
+      end
+    elsif args.nil? && (args.is_a? Class)
+      return false
+    elsif !args.nil? && (args.is_a? Class)
+      return true
+    elsif Regexp
+      my_each do |i|
+        return false if args.match(i)
+      end
+    end
+    true
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  def my_any?(args = nil)
+    check = false
+    is_true = false
+    if block_given?
+      my_each do |i|
+        if yield(i) == true
+          check = true
+          return check
+        else
+          check = false
+        end
+      end
+      return check
+    elsif args.class == Regexp
+      my_each do |i|
+        return true if i.match(args)
+
+        return false
+      end
+    elsif !args.nil? && (args.is_a? Class)
+      my_each do |i|
+        if i.is_a? args
+          is_true = true
+          return true
+        else
+          is_true = false
+        end
+      end
+      return is_true
+    end
+    unless block_given?
+      my_each { |x| return true if x == true }
+      return false
+    end
+    false
+  end
+
+  def my_none?(args = nil)
+    if block_given?
+      check = false
+      my_each do |element|
+        if yield(element) == false
+          check = true
+        else
+          check = false
+          return check
+        end
+      end
+      return check
+    elsif !args.nil? && (args.is_a? Class)
+      my_each do |i|
+        return false if i.is_a? args
+      end
+    elsif args.class == Regexp
+      my_each do |i|
+        return false if i.match(args)
+
+        return true
+      end
+    end
+    unless block_given?
+      my_each { |x| return false if x == true }
+      return true
+    end
+    false
+  end
+
+  # rubocop:enable Metrics/MethodLength
+  # rubocop:enable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+  # My Count
+  def my_count(num = 0)
```

Comment:
- [ ] returns the number of items in enum through enumeration
```ruby
array = Array.new(100){rand(0...9)}
array.my_count == array.count # should return true
```
- [ ] returns false if the block ever returns true for all elements::range
```ruby
range = Range.new(5,50)
range.my_none?(&false_block) == range.none?(&false_block) # should return true
```
- [ ] counts the number of items in enum that are equal to item if an argument is given
```ruby
array.my_count(0) == array.count(0) # should return true
```